### ISSUE TITLE
Call of Duty: Road to Victory (Seemingly incorrect address loads?)

### ISSUE THREAD

**Issue by EdnessP** [hrydgard/ppsspp#15455]
### Game or games this happens in

ULES-00643, ULES-00644, ULUS-10218 - Call of Duty: Roads to Victory

### What area of the game / PPSSPP

A year ago or so I wrote [this page](https://tcrf.net/Call_of_Duty:_Roads_to_Victory), only testing it on PPSSPP where it worked, and a friend offered to test it on a real PSP, which oddly just flat out didn't work.  Few days ago I got a PSP of my own and could investigate it, and it looks like the addresses it loads are off by 0x48000.  It's not a matter of the addresses themselves being set randomly, as the functions load a static address to a register.

Take this for example, where it loads 0x08B65EA8 into $a1 in both cases, on PPSSPP that turns into 0x08E38024 (relocation?) but on a PSP while I can't exactly see the $a1 register, I've worked out that it loads 0x08E80024, a difference of exactly 0x48000.  Changing all of the codes on the linked page earlier to add 0x48000 to all the addresses makes it work on a real PSP, but PPSSPP on both Windows and Android won't work then.
![PPSSPP Debugger](https://media.discordapp.net/attachments/340499300754915329/955217434825097306/unknown.png)
![PSP Disassembler](https://cdn.discordapp.com/attachments/340499300754915329/955216112155500575/20220320_232725.jpg)

This doesn't seem to be an issue with the extra RAM that can be allocated from UMD cache on 2K and other models either, I've tested it with the extra memory both enabled and disabled on my PSP, and testing codes I wrote for some other games originally on PPSSPP now on a PSP worked just fine without changing the addresses.

### What should happen

The addresses shouldn't mismatch between a real PSP and an emulated system, making patches incompatible between each other.

### Platform

Windows

### Mobile phone model or graphics card

Samsung Galaxy Note9, NVIDIA GeForce GTX 1080

### PPSSPP version affected

v1.10.? - v1.12.3-1007-ga155c4972

### Graphics backend (3D API)

Vulkan

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [ ] Try without any cheats and without loading any save states.
- [ ] Include logs or screenshots of issue.

**Comment by hrydgard** [comment]
Yeah, memory allocations being somewhat off vs the real hardware is a long standing issue. We do high level emulation of PSP system libraries and some of them do internal allocations that we don't really know about...
